src/optimization-engine/Deploy-AzureOptimizationEngine.ps1

```diff
@@ -0,0 +1,1532 @@
+            $laWorkspaceName = $deploymentOptions["WorkspaceName"]
+        }
+        else {
+            $laWorkspaceName = $laWorkspaceNameTemplate -f $namePrefix
+        }
+        $sqlDatabaseName = "azureoptimization"
+    }
+    else {
+        $resourceGroupName = $deploymentOptions["ResourceGroupName"]
+        $storageAccountName = $deploymentOptions["StorageAccountName"]
+        $automationAccountName = $deploymentOptions["AutomationAccountName"]
+        $sqlServerName = $deploymentOptions["SqlServerName"]
+        $sqlDatabaseName = $deploymentOptions["SqlDatabaseName"]        
+        $laWorkspaceName = $deploymentOptions["WorkspaceName"]        
+        $deploymentName = $deploymentNameTemplate -f $resourceGroupName
+    }
+}
+#endregion
+
+#region Resource naming availability checks
+Write-Host "Checking name prefix availability..." -ForegroundColor Green
+
+Write-Host "...for the Storage Account..." -ForegroundColor Green
+$sa = Get-AzStorageAccount -ResourceGroupName $resourceGroupName -Name $storageAccountName -ErrorAction SilentlyContinue
+if ($null -eq $sa) {
+    $saNameResult = Get-AzStorageAccountNameAvailability -Name $storageAccountName
+    if (-not($saNameResult.NameAvailable)) {
+        $nameAvailable = $false
+        Write-Host "$($saNameResult.Message)" -ForegroundColor Red
+    }    
+}
+else {
+    Write-Host "(The Storage Account was already deployed)" -ForegroundColor Green
+}
+
+if ("N", "n" -contains $workspaceReuse) {
+    Write-Host "...for the Log Analytics workspace..." -ForegroundColor Green
+
+    $logAnalyticsReuse = $false
+    $laWorkspaceResourceGroup = $resourceGroupName
+
+    $la = Get-AzOperationalInsightsWorkspace -ResourceGroupName $resourceGroupName -Name $laWorkspaceName -ErrorAction SilentlyContinue
+    if ($null -eq $la) {
+        $laNameResult = Invoke-WebRequest -Uri "https://portal.loganalytics.io/api/workspaces/IsWorkspaceExists?name=$laWorkspaceName"
+        if ($laNameResult.Content -eq "true") {
+            $nameAvailable = $false
+            Write-Host "The Log Analytics workspace $laWorkspaceName is already taken." -ForegroundColor Red
+        }
+    }
+    else {
+        Write-Host "(The Log Analytics Workspace was already deployed)" -ForegroundColor Green
+    }
+}
+else {
+    $logAnalyticsReuse = $true
+}
+
+Write-Host "...for the Azure SQL Server..." -ForegroundColor Green
+$sql = Get-AzSqlServer -ResourceGroupName $resourceGroupName -Name $sqlServerName -ErrorAction SilentlyContinue
+if ($null -eq $sql -and -not($sqlServerName -like "*.database.*") -and -not($IgnoreNamingAvailabilityErrors)) {
+
+    $SqlServerNameAvailabilityUriPath = "/subscriptions/$subscriptionId/providers/Microsoft.Sql/checkNameAvailability?api-version=2014-04-01"
+    $body = "{`"name`": `"$sqlServerName`", `"type`": `"Microsoft.Sql/servers`"}"
+    $sqlNameResult = (Invoke-AzRestMethod -Path $SqlServerNameAvailabilityUriPath -Method POST -Payload $body).Content | ConvertFrom-Json
```

Comment:
```suggestion
    # TODO: Switch to custom Invoke-Rest command to leverage telemetry tracking
    $sqlNameResult = (Invoke-AzRestMethod -Path $SqlServerNameAvailabilityUriPath -Method POST -Payload $body).Content | ConvertFrom-Json
```

Reply:
I need guidance on how to leverage telemetry tracking and why switching to Invoke-Rest is necessary in this context.